server/service/service_teams.go

```diff
@@ -63,6 +63,14 @@ func (svc *Service) ListTeams(ctx context.Context, opt fleet.ListOptions) ([]*fl
 	return nil, fleet.ErrMissingLicense
 }
 
+func (svc *Service) ListAvailableTeamsForUser(ctx context.Context, user *fleet.User) ([]*fleet.Team, error) {
+	// skipauth: No authorization check needed due to implementation returning
+	// only license error.
+	svc.authz.SkipAuthorization(ctx)
+
+	return nil, fleet.ErrMissingLicense
```

Comment:
I believe this will make all `login`/`get_user` requests fail in the Fleet free version. 

Maybe in the two places this called in we could check for `errors.Is(err, fleet.ErrMissingLicense)` and return empty `available_teams` in that case.